internal/updater/extract.go

```diff
@@ -23,55 +24,94 @@ func extractFile(buf []byte, filename, dest string) error {
 		mode = fi.Mode()
 	}
 
+	if err := os.Remove(dest); err != nil {
+		if !os.IsNotExist(err) {
+			return fmt.Errorf("unable to remove destination file: %q", err)
+		}
+	}
+
+	dfh, err := os.OpenFile(dest, os.O_WRONLY|os.O_CREATE|os.O_EXCL, mode)
+	if err != nil {
+		return errors.Wrapf(err, "Failed to open file: %s", dest)
+	}
+	defer func() {
+		_ = dfh.Close()
```

Comment:
Since this is a writable file it would be best to check Close for errors as these can happen upon closing with the sync and all.

Reply:
Yes ...